src/control/tools/EditSelection.cpp

```diff
@@ -690,17 +704,134 @@ void EditSelection::moveSelection(double dx, double dy) {
     this->snappedBounds.x += dx;
     this->snappedBounds.y += dy;
 
-    ensureWithinVisibleArea();
-
     updateMatrix();
 
     this->view->getXournal()->repaintSelection();
 }
 
+void EditSelection::setEdgePan(bool pan) {
+    if (pan && this->edgePanHandler == nullptr) {
+        this->edgePanHandler = g_timeout_source_new(1000 / 30);
+        g_source_set_callback(this->edgePanHandler, reinterpret_cast<int (*)(void*)>(EditSelection::handleEdgePan),
```

Comment:
Why not simply like this?
```suggestion
        g_source_set_callback(this->edgePanHandler, reinterpret_cast<GSourceFunc>(EditSelection::handleEdgePan),
```

Reply:
GSourceFunc is not available until glib 2.58.

Reply:
You probably mean [`G_SOURCE_FUN`](https://developer.gnome.org/glib/stable/glib-The-Main-Event-Loop.html#G-SOURCE-FUNC:CAPS). We use `reinterpret_cast<GSourceFunc>` already in various other source files.

Reply:
You're right. Fixed.